data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java

```diff
@@ -0,0 +1,262 @@
+  private static final String TOPIC =
+      format("%s%s-%s", TOPIC_PREFIX, BROKER_NAMESPACE, BROKER_NAME);
+  private static final int NUM_BROKERS = 1;
+  private static final int ZK_PORT = 2181;
+  private static final int KAFKA_PORT = 9092;
+  private static final int NUM_PARTITIONS = 10;
+  private static final int REPLICATION_FACTOR = 1;
+  private static final int INGRESS_PORT = 12345;
+  private static final int SERVICE_PORT = INGRESS_PORT + 1;
+
+  private static final String TYPE_CE_1 = "type-ce-1";
+  private static final String TYPE_CE_2 = "type-ce-2";
+  private static final String PATH_SERVICE_1 = "/service-1";
+  private static final String PATH_SERVICE_2 = "/service-2";
+
+  static {
+    assertThat(PATH_SERVICE_1).isNotEqualTo(PATH_SERVICE_2);
+  }
+
+  private static File dataDir;
+  private static KafkaCluster kafkaCluster;
+  private static BrokersManager<CloudEvent> brokersManager;
+
+  @BeforeAll
+  public static void setUp(final Vertx vertx, final VertxTestContext context) throws IOException {
+    setUpKafkaCluster();
+    setUpReceiver(vertx, context);
+    setUpDispatcher(vertx);
+  }
+
+  @Test
+  public void execute(final Vertx vertx, final VertxTestContext context) {
+
+    final var checkpoints = context.checkpoint(3);
+
+    // event written by the source to the Broker
+    final var requestEvent = CloudEventBuilder.v1()
+        .withId(UUID.randomUUID().toString())
+        .withDataSchema(URI.create("/api/data-schema-ce-1"))
+        .withSource(URI.create("/api/rossi"))
+        .withSubject("subject-ce-1")
+        .withData("data-ce-1".getBytes())
+        .withType(TYPE_CE_1)
+        .build();
+
+    // event sent in the response by the Callable service
+    final var responseEvent = CloudEventBuilder.v03()
+        .withId(UUID.randomUUID().toString())
+        .withDataSchema(URI.create("/api/data-schema-ce-2"))
+        .withSubject("subject-ce-2")
+        .withSource(URI.create("/api/rossi"))
+        .withData("data-ce-2".getBytes())
+        .withType(TYPE_CE_2)
+        .build();
+
+    // reconcile brokers/triggers
+    brokersManager.reconcile(Map.of(
+        new BrokerWrapper(Broker.newBuilder()
+            .setTopic(TOPIC)
+            .setId(UUID.randomUUID().toString())
+            .build()),
+        Set.of(
+            new TriggerWrapper(Trigger.newBuilder()
+                .setDestination(format("http://localhost:%d%s", SERVICE_PORT, PATH_SERVICE_1))
+                .putAttributes(ContextAttributes.TYPE.name().toLowerCase(), TYPE_CE_1)
+                .setId(UUID.randomUUID().toString())
+                .build()),
+            new TriggerWrapper(Trigger.newBuilder()
+                .setDestination(format("http://localhost:%d%s", SERVICE_PORT, PATH_SERVICE_2))
+                .putAttributes(ContextAttributes.TYPE.name().toLowerCase(), TYPE_CE_2)
+                .setId(UUID.randomUUID().toString())
+                .build())
+        )
+    ));
```

Comment:
you need to add a latch here or you execute the rest of the test in the callback of reconcile, otherwise you're not waiting for the reconcile to complete, possibly making this test flaky